### ISSUE TITLE
Support iterating over baggage items

### ISSUE THREAD

**Issue by bok-at-built** [DataDog/dd-trace-js#2409]
Hi, I noticed the opentracing spec states the following in regards to a `SpanContext`:

`"...semantically the caller should be able to efficiently iterate through all baggage items in one pass given a SpanContext instance"` ([ref](https://github.com/opentracing/specification/blob/master/specification.md#iterate-through-all-baggage-items)) 

I don't see a way to do this with datadog's tracer implementation. Am I misunderstanding the spec or is this a missing feature in `dd-trace`?

For context, I'd like this feature so downstream services to be able to iterate over the arbitrary key-values pairs passed to them in the baggage and then add them as span tags or log fields. The alternative is I'd need to make every node in our service mesh aware of its upstreams' baggage shape(s), which seems untenable. Thanks in advance!

**Comment by rochdev** [comment]
This doesn't seem to be supported in the [JavaScript implementation](https://github.com/opentracing/opentracing-javascript/blob/master/src/span.ts). We could always add it anyway since we have more functionalities that what OpenTracing offers in the first place, but it wouldn't be compatible with other implementations (which is the whole point of using OpenTracing).

**Comment by bok-at-built** [comment]
Ah gotcha, thanks for the comment @rochdev. That's weird :/ The spec mentions it in plain english and I even see it in the [golang spec](https://github.com/opentracing/opentracing-go/blob/master/span.go#L21).

I guess we can just close this out and wait for dd to move to open telemetry, which I see implements this functionality ([ref](https://github.com/open-telemetry/opentelemetry-js/blob/main/api/src/baggage/types.ts#L69)). In the meantime we'll just unsafely access `span.context()._baggageItems`.
